packages/react-reconciler/src/__tests__/ReactNewContext-test.internal.js

```diff
@@ -762,6 +762,45 @@ describe('ReactNewContext', () => {
     expect(ReactNoop.getChildren()).toEqual([span('Child')]);
   });
 
+  it('renders context children when child is an instance method', () => {
+    const Context = React.createContext(0);
+
+    class App extends React.Component {
+      state = {
+        value: 0,
+      };
+
+      renderConsumer = (context) => {
+        ReactNoop.yield('App#renderConsumer');
+        return <span prop={this.state.value} />;
```

Comment:
@gaearon The issue is not so much about context propagating, but that this instance method that reads from `state` to be rendered and updated when `setState` is called.

Reply:
The following is pretty close to actual code that experienced this.

```
// @flow

import * as React from 'react';
import { login } from 'api';
import { BetaFeatureConsumer } from 'BetaFeatures';
import { SingleSignOn } from 'SingleSignOn';

type P = {};
type S = {
  username: string,
  password: string,
  loading: boolean,
  errors: *,
};

export default class Login extends React.Component<P, S> {
  state = {
    username: '',
    password: '',
    loading: false,
    errors: null,
  };

  onChange = (event: SyntheticEvent<HTMLInputElement>) => {
    const { name, value } = event.currentTarget;
    this.setState({ [name]: value });
  };

  onSubmit = (event: SyntheticEvent<*>) => {
    event.preventDefault();
    this.setState({ loading: true, errors: null });
    login(this.state.username, this.state.password).then(data => {
      this.setState({ loading: false });
      if (data.hasError) {
        this.setState({ errors: { ...this.state.errors, ...data.errors } });
      }
    });
  };

  _render = v => {
    return (
      <form onSubmit={this.onSubmit}>
        <input
          label="Username"
          name="username"
          onChange={this.onChange}
          value={this.state.username}
        />
        <input
          label="Password"
          name="password"
          onChange={this.onChange}
          value={this.state.password}
          type="password"
        />
        <button type="submit" disabled={this.state.loading}>
          Submit
        </button>
        {v.singleSignOn && <SingleSignOn onComplete={this.onSubmit} />}
      </form>
    );
  };

  render() {
    return (
      <BetaFeatureConsumer>
        {this._render}
      </BetaFeatureConsumer>
    );
  }
}
```